sources/api/migration/migrations/v1.11.0/aws-creds/src/main.rs

```diff
@@ -0,0 +1,35 @@
+#![deny(rust_2018_idioms)]
+
+use migration_helpers::common_migrations::AddSettingsMigration;
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// We added new settings for AWS credential configuration.
+fn run() -> Result<()> {
+    if cfg!(variant_platform = "aws") {
+        migrate(AddSettingsMigration(&[
+            "settings.aws.config",
+            "settings.aws.credentials",
+            "settings.aws.profile",
+        ]))
+    } else {
+        // Non-AWS variants did not have the region setting until this point,
+        // so need to be included here to remove on downgrade.
+        migrate(AddSettingsMigration(&[
+            "settings.aws.config",
+            "settings.aws.credentials",
+            "settings.aws.profile",
+            "settings.aws.region",
+        ]))
+    }
```

Comment:
@etungsten do you know if this branch needs to be an `AddPrefixesMigration` across all of `settings.aws` in order to clean up the "aws" directory in the datastore?

(It might not hurt to do that just in case.)

Reply:
I was actually wondering about that for the non-AWS case.